### PR TITLE
Fix MultiHeadAttention docs

### PR DESCRIPTION
**Pull request by cgarciae**
Docs on the site look weird due to formatting issues: https://www.tensorflow.org/addons/api_docs/python/tfa/layers/MultiHeadAttention

This PR tries to fix the formatting and also improves parts of the text a bit.

### CODE REVIEW

File: tensorflow_addons/layers/multihead_attention.py [hunk 1]
```diff
@@ -21,11 +21,11 @@
 @tf.keras.utils.register_keras_serializable(package="Addons")
 class MultiHeadAttention(tf.keras.layers.Layer):
     r"""
-    MultiHead Attention layer.
+    MultiHead Attentsion layer.
```
Comment (seanpmorgan) [inline tensorflow_addons/layers/multihead_attention.py:24]:
```suggestion
    MultiHead Attention layer.
```

File: tensorflow_addons/layers/multihead_attention.py [hunk 2]
```diff
@@ -70,19 +70,16 @@ class MultiHeadAttention(tf.keras.layers.Layer):
         bias_regularizer: regularizer, regularizer for the bias weights.
         bias_constraint: constraint, constraint for the bias weights.
 
-    Call Arguments
-        inputs:  List of the following tensors:
-            * `query`: Tensor of shape `(..., query_elements, query_depth)`
-            * `key`: `Tensor of shape '(..., key_elements, key_depth)`
-            * `value`: Tensor of shape `(..., key_elements, value_depth)` (optional)
+    Call Arguments:
+        inputs:  List of the tensors `[query, key, value]` where  `query` has shape `(..., query_elements, query_depth)`, `key` has shape '(..., key_elements, key_depth)`, and `value` has shape `(..., key_elements, value_depth)`. `value` is optional, if not given `key` will be used.
```
Comment (WindQAQ) [inline tensorflow_addons/layers/multihead_attention.py:74]: Maybe we can leave listings `-` alone.
https://github.com/tensorflow/tensorflow/blob/v2.2.0/tensorflow/python/keras/layers/pooling.py#L183-L187
https://www.tensorflow.org/api_docs/python/tf/keras/layers/AveragePooling1D

File: tensorflow_addons/layers/multihead_attention.py [hunk 3]
```diff
@@ -70,19 +70,16 @@ class MultiHeadAttention(tf.keras.layers.Layer):
...
         `1` indicates attention and `0` indicates no attention.
 
-    Output shape
-        - `(..., query_elements, output_size)` if `output_size` is given, else
-        - `(..., query_elements, value_depth)` if `value` is given, else
-        - `(..., query_elements, key_depth)`
+    Output shape:
+        `(..., query_elements, output_size)` if `output_size` is given, else
+        `(..., query_elements, value_depth)` if `value` is given, else
+        `(..., query_elements, key_depth)`
```
Comment (WindQAQ) [inline tensorflow_addons/layers/multihead_attention.py:82]: ditto

File: tensorflow_addons/layers/multihead_attention.py (hunk 2 above)
Comment (seanpmorgan) [reply tensorflow_addons/layers/multihead_attention.py:74]: Yes I would prefer we stick to what's being done elsewhere in the ecosystem. I think if we copy the formatting as linked above it comes out quite readable in the generated docs. What do you think @cgarciae ?

**Comment by cgarciae** [comment]
I'll go back to the listing style if it works, thought it might screw the format.

**Comment by cgarciae** [comment]
I think this looks better, is there a way to check how the page will look now?

**Comment by WindQAQ** [comment]
> I think this looks better, is there a way to check how the page will look now?

Hi, maybe you can follow the instructions here.
https://github.com/tensorflow/addons/tree/master/docs

**Pull request merged**